### PR TITLE
Fix UI blocking on HCaptcha creation

### PR DESCRIPTION
**Pull request by CAMOBAP**
Also better error handling if Webview is terminated, e.g. in scenarios like #103

### CODE REVIEW

**Comment by e271828-** [comment]
Notes: waiting on changes for safer state machine behavior.

File: HCaptcha/Classes/HCaptchaWebViewManager.swift [hunk 1]
```diff
@@ -77,6 +76,11 @@ internal class HCaptchaWebViewManager {
         }
     }
 
+    fileprivate var didLoadTimer: Timer?
+
+    /// flab which will prevent
```
Comment (e271828-) [inline HCaptcha/Classes/HCaptchaWebViewManager.swift:81]: missing word?

**Comment by CAMOBAP** [comment]
Need to be updated after https://github.com/hCaptcha/HCaptcha-ios-sdk/pull/95 merge

**Comment by e271828-** [comment]
@CAMOBAP is this PR still the approach you want to take?

**Comment by CAMOBAP** [comment]
@e271828- Probably, let me review it myself, because I already forgot the implementation details

File: HCaptcha/Classes/HCaptchaWebViewManager.swift [hunk 2]
```diff
@@ -385,4 +367,24 @@ extension HCaptchaWebViewManager: WKNavigationDelegate {
...
         decisionHandler(WKNavigationActionPolicy.allow)
     }
+
+    func webView(_ webView: WKWebView, didFail navigation: WKNavigation!, withError error: Error) {
+        completion?(HCaptchaResult(error: .unexpected(error)))
+    }
+
+    func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {
+        completion?(HCaptchaResult(error: .unexpected(error)))
+    }
```
Comment (e271828-) [inline HCaptcha/Classes/HCaptchaWebViewManager.swift:377]: This looks a little weird. Also, please add comments to these functions.

Comment (CAMOBAP) [reply HCaptcha/Classes/HCaptchaWebViewManager.swift:377]: To give more context, navigation happens each time when we load content in web view and I think we must catch those issue too as unexpected.

I will add comments

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -116,15 +120,17 @@ jobs:
...
     name: Build Samples
     needs: sdk
-    runs-on: macos-11
+    runs-on: macos-12
     continue-on-error: ${{ matrix.experimental }}
     strategy:
       fail-fast: false
       matrix:
-        xcode: [ '12.4' ]
+        xcode: [ '13.1' ]
```
Comment (DSergiu) [inline .github/workflows/workflow.yml:128]: Should we use the `${{ env.XCODE_VERSION }}` here?

Comment (CAMOBAP) [reply .github/workflows/workflow.yml:128]: That's make sens, I will update PR

**Review by e271828- — APPROVED** [review]

**Pull request merged**